detox/src/client/actions/SyncStatusFormatter.js

```diff
@@ -0,0 +1,277 @@
+const Ajv = require('ajv');
+
+const DetoxRuntimeError = require('../../errors/DetoxRuntimeError');
+
+class SyncStatusFormatter {
+  static formatJSONStatus(jsonStatus) {
+    if (!SyncStatusFormatter.isValidJSONStatus(jsonStatus)) {
+      throw new DetoxRuntimeError(`Given sync status is not compatible with the status schema, ` +
+        `given status: ${JSON.stringify(jsonStatus)}`);
+    }
+
+    if (SyncStatusFormatter.isAppIdle(jsonStatus)) {
+      return 'The app seems to be idle';
+    }
+
+    if (jsonStatus.busy_resources === undefined) {
+      throw new DetoxRuntimeError(`Given sync status is invalid, app is busy but busy-resources are not defined`);
+    }
+    const resourcesDescriptions = SyncStatusFormatter.resourcesDescriptionsFromJSON(jsonStatus.busy_resources);
+    return `The app is busy with the following tasks:\n${resourcesDescriptions.join('\n')}`;
+  }
+
+  static isValidJSONStatus(jsonStatus) {
+    const ajv = new Ajv();
+    const isValidStatus = ajv.compile(SyncStatusFormatter.jsonStatusSchema());
+    return isValidStatus(jsonStatus);
+  }
+
+  static jsonStatusSchema() {
+    return {
+      type: 'object',
+      properties: {
+        app_status: { 'enum': ['busy', 'idle'] },
+        busy_resources: {
+          type: 'array',
+          items: {
+            type: 'object',
+            properties: {
+              name: { 'type': 'string' },
+              description: {
+                type: 'object',
+                additionalProperties: true
```

Comment:
Why won't you use JSON schema deeper for this object?
https://stackoverflow.com/questions/35450405/how-to-represent-sum-union-types-in-in-json-schema

Reply:
mmm why not.. that's indeed better, I'll do it
also thinking about this we can enforce `busy_resource` to be non-empty from the schema and not from the